src/classes/yaml-utils.ts

```diff
@@ -106,14 +152,18 @@ export default class YamlUtils {
    * @param file The JSON file that is being edited to create a .yaml file
    * @returns the edited JSON file for service
    */
-  private static setupService = (args: any, file: any): any => {
+  private static setupService = (args: KubeArgs, namespace: string, file: any): any => {
+
+    file.metadata.name = args.name;
+    file.metadata.labels.run = namespace;
+
     if (args.ports) {
       file.spec.ports = [];
       for (const port in args.ports) {
         file.spec.ports.push(port);
       }
     }
-    file.spec.selector.app = args.name;
+    file.spec.selector.app = namespace;
```

Comment:
are you sure this should be a namespace?

Reply:
Fixed syntaxes, now pod is created directly under deployment, pod now listens to created service, created service now attempts to match pods within same namespace (project name)